src/PayID/pay-id-client.ts

```diff
@@ -24,53 +18,41 @@ export default class PayIDClient {
    * @returns An XRP address representing the given PayID if one exists, otherwise undefined.
    */
+  // TODO(keefertaylor): It's likely that at some point we'll need to store instance state in this class. Make this a non-static
+  //                     method until a complete API spec proves it can be static.
+  // eslint-disable-next-line class-methods-use-this
   public async xrpAddressForPayID(payID: string): Promise<string | undefined> {
     const paymentPointer = PayIDUtils.parsePaymentPointer(payID)
     if (!paymentPointer) {
       throw PayIDError.invalidPaymentPointer
     }
 
-    const url = `https://${paymentPointer.host}${paymentPointer.path}`
-    // TODO(keefertaylor): Attach network parameter.
-    // TODO(keefertaylor): Generalize the below to allow requesting in other types.
-    const requestConfig = {
-      headers: {
-        Accept: 'application/xrp+json',
-      },
-    }
+    // Swagger generates the '/' in the URL by default and the payment pointer's 'path' is prefixed by a '/'. Strip off the leading '/'.
+    const path = paymentPointer.path.substring(1)
 
-    let response
-    try {
-      response = await this.axiosInstance.get<{ address: string }>(
-        url,
-        requestConfig,
-      )
-    } catch (error) {
-      // Handle erroneous responses from the server.
-      if (error.response) {
-        // 404 means no mapping was found.
-        if (error.response.status === 404) {
-          return undefined
-        }
+    // TODO(keefertaylor): Append headers here.
+    const client = new ApiClient()
+    client.basePath = `https://${paymentPointer.host}`
+    const apiInstance = new DefaultApi(client)
 
-        // Otherwise re-throw the error as an unexepected response.
-        throw new PayIDError(
-          PayIDErrorType.UnexpectedResponse,
-          `${error.response.status} ${error.response.statusText}: ${error.response.data}`,
-        )
-      } else {
-        // Generically errors from the server which contained no response (timeouts, etc).
-        throw new PayIDError(PayIDErrorType.Unknown, error.message)
-      }
-    }
-
-    const address = response.data?.address
-    if (!address) {
-      throw new PayIDError(
-        PayIDErrorType.UnexpectedResponse,
-        'Sucessful response was in an unknown format',
-      )
-    }
-    return address
+    return new Promise((resolve, reject) => {
+      // Ignore unused var in generated code.
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      apiInstance.resolvePayID(path, (error, data, _response) => {
+        if (error) {
+          if (error.status === 404) {
+            // Not Found
+            resolve(undefined)
+          } else {
+            const message = `${error.status}: ${error.response.text}`
+            reject(new PayIDError(PayIDErrorType.UnexpectedResponse, message))
+          }
+        } else if (data?.address) {
+          resolve(data.address)
```

Comment:
What does this `resolve(data.address)` call do?

Reply:
This resolves the promise with the XRP address that came back. Something else you're looking for?

Reply:
No that seems fine. Based on the current public API spec: https://docs.google.com/document/d/1Xc3dgjFhG_XYu_slMQrWQZhJeNzb2hMzhvA-Gw_N9Ug/edit#

This might be `data?.addressDetails?.address` for XRP addresses.